website/docs/internals/resource-addressing.html.markdown

```diff
@@ -38,15 +50,17 @@ resource_type.resource_name[resource index]
 
  * `resource_type` - Type of the resource being addressed.
  * `resource_name` - User-defined name of the resource.
- * `[resource index]` - an optional index into a resource with multiple
+ * `[resource index]` - (Optional) Index into a resource with multiple
    instances, surrounded by square brace characters (`[` and `]`).
 
 -> In Terraform v0.12 and later, a resource spec without a module path prefix
 matches only resources in the root module. In earlier versions, a resource spec
 without a module path prefix will match resources with the same type and name
 in any descendent module.
 
-__Resource index__:
+__Index values for Modules and Resources__:
```

Comment:
I like that we'd have a shared heading for both of these, rather than duplicating the content. Should we link from the sections above (perhaps on the word "Index" in the list of addressing components?) down to this section to help someone who might only be reading the "module path" section know that there's more detail about it further down?

Reply:
Will link!